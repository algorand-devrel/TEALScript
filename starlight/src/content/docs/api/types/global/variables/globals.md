---
editUrl: false
next: false
prev: false
title: "globals"
---

> **`const`** **globals**: `Object`

## Type declaration

### assetCreateMinBalance

> **assetCreateMinBalance**: [`uint64`](../type-aliases/uint64.md)

### assetOptInMinBalance

> **assetOptInMinBalance**: [`uint64`](../type-aliases/uint64.md)

### callerApplicationAddress

> **callerApplicationAddress**: [`Address`](../classes/Address.md)

### callerApplicationID

> **callerApplicationID**: [`Application`](../classes/Application.md)

### creatorAddress

> **creatorAddress**: [`Address`](../classes/Address.md)

### currentApplicationAddress

> **currentApplicationAddress**: [`Address`](../classes/Address.md)

### currentApplicationID

> **currentApplicationID**: [`Application`](../classes/Application.md)

### genesisHash

> **genesisHash**: [`bytes32`](../type-aliases/bytes32.md)

### groupID

> **groupID**: [`bytes32`](../type-aliases/bytes32.md)

### groupSize

> **groupSize**: [`uint64`](../type-aliases/uint64.md)

### latestTimestamp

> **latestTimestamp**: [`uint64`](../type-aliases/uint64.md)

### logicSigVersion

> **logicSigVersion**: [`uint64`](../type-aliases/uint64.md)

### maxTxnLife

> **maxTxnLife**: [`uint64`](../type-aliases/uint64.md)

### minBalance

> **minBalance**: [`uint64`](../type-aliases/uint64.md)

### minTxnFee

> **minTxnFee**: [`uint64`](../type-aliases/uint64.md)

### opcodeBudget

> **opcodeBudget**: [`uint64`](../type-aliases/uint64.md)

### round

> **round**: [`uint64`](../type-aliases/uint64.md)

### zeroAddress

> **zeroAddress**: [`Address`](../classes/Address.md)

## Source

[types/global.d.ts:800](https://github.com/algorandfoundation/tealscript/blob/18ba30a9/types/global.d.ts#L800)
