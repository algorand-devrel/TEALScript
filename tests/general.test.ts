/* eslint-disable func-names */
/* eslint-disable prefer-arrow-callback */

import * as algokit from '@algorandfoundation/algokit-utils';
import { describe, test } from '@jest/globals';
import { artifactsTest, compileAndCreate, runMethod, algodClient, kmdClient } from './common';

const NAME = 'GeneralTest';
const PATH = 'tests/contracts/general.algo.ts';
const ARTIFACTS_DIR = 'tests/contracts/artifacts/';

describe('General', function () {
  artifactsTest(PATH, ARTIFACTS_DIR, NAME);

  describe('E2E', function () {
    const sender = algokit.getLocalNetDispenserAccount(algodClient, kmdClient);

    test('staticArray', async function () {
      const { appClient } = await compileAndCreate(await sender, PATH, ARTIFACTS_DIR, NAME);
      await runMethod({ appClient, method: 'txnTypeEnum' });
    });
  });
});
