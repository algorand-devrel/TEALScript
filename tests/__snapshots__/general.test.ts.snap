// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`General GeneralTest Artifacts Maintains program size 1`] = `4461`;
